astropy/utils/data_info.py

```diff
@@ -60,8 +60,10 @@ def serialize_context_as(context):
     """
     old_context = BaseColumnInfo._serialize_context
     BaseColumnInfo._serialize_context = context
-    yield
-    BaseColumnInfo._serialize_context = old_context
+    try:
```

Comment:
Slightly out of scope, but the original code was not restoring context after an exception (which I noticed in testing).

Reply:
:+1: